kolibri/plugins/coach/assets/src/modules/classSummary/dataHelpers.js

```diff
@@ -107,6 +118,18 @@ export default {
       return lesson.assignments.length ? this.getLearnersForGroups(lesson.groups) : [];
     };
   },
+  /*
+   * Return array of names of groups followed by names of assigned
+   * ad hoc learners
+   */
+  getRecipientNamesForLesson(state) {
+    return function(lesson) {
+      const assignments = lesson.lesson_assignments.map(l => l.collection);
+      return this.getGroupNames(state.lessonMap[lesson.id].groups).concat(
+        this.getAdHocLearners(assignments).map(learnerId => state.learnerMap[learnerId].name)
```

Comment:
Maybe consider a `getAdHocLearnerNames` helper since this is done more than once?